File: src/main/java/com/hospital/hospital_billing_system/doctor/service/DoctorService.java
package com.hospital.hospital_billing_system.doctor.service;

import com.hospital.hospital_billing_system.doctor.dto.DoctorConsultationChargeRequest;
import com.hospital.hospital_billing_system.doctor.dto.DoctorConsultationChargeResponse;
import com.hospital.hospital_billing_system.doctor.dto.DoctorRequest;
import com.hospital.hospital_billing_system.doctor.dto.DoctorResponse;

import java.util.List;
import java.util.UUID;

/**
 * Business service defining operations for Doctor and Provider credentialing.
 */
public interface DoctorService {

    DoctorResponse registerDoctor(DoctorRequest request);

    DoctorResponse getDoctorById(UUID doctorId, UUID tenantId);

    DoctorResponse getDoctorByProviderNumber(String providerNo);

    List<DoctorResponse> getDoctorsByTenant(UUID tenantId);

    List<DoctorResponse> getDoctorsByDepartment(UUID tenantId, UUID departmentId);

    DoctorResponse updateDoctor(UUID doctorId, UUID tenantId, DoctorRequest request);

    void deleteDoctor(UUID doctorId, UUID tenantId);

    // Method to validate practitioner credentials and calculate consultation charge payload for Central Billing
    DoctorConsultationChargeResponse verifyAndCalculateConsultationCharge(DoctorConsultationChargeRequest request);


}

File: frontend/src/services/doctorService.js
import apiClient from "./apiClient";

const doctorService = {
  getAllDoctors: (tenantId) => {
    return apiClient.get("/api/v1/doctors", {
      headers: {
        "X-Tenant-ID": tenantId,
      },
    });
  },

  getDoctorById: (doctorId, tenantId) => {
    return apiClient.get(`/api/v1/doctors/${doctorId}`, {
      headers: {
        "X-Tenant-ID": tenantId,
      },
    });
  },

  getDoctorByProviderNumber: (providerNo) => {
    return apiClient.get("/api/v1/doctors/by-provider-number", {
      params: {
        providerNo,
      },
    });
  },

  getDoctorsByDepartment: (departmentId, tenantId) => {
    return apiClient.get(
      `/api/v1/doctors/department/${departmentId}`,
      {
        headers: {
          "X-Tenant-ID": tenantId,
        },
      }
    );
  },

  createDoctor: (doctorData) => {
    return apiClient.post("/api/v1/doctors", doctorData);
  },

  updateDoctor: (doctorId, tenantId, doctorData) => {
    return apiClient.put(
      `/api/v1/doctors/${doctorId}`,
      doctorData,
      {
        headers: {
          "X-Tenant-ID": tenantId,
        },
      }
    );
  },

  deleteDoctor: (doctorId, tenantId) => {
    return apiClient.delete(
      `/api/v1/doctors/${doctorId}`,
      {
        headers: {
          "X-Tenant-ID": tenantId,
        },
      }
    );
  },

  getConsultationCharge: (requestData) => {
    return apiClient.post(
      "/api/v1/doctors/consultation-charge",
      requestData
    );
  },
};

export default doctorService;